samples: matter: Unified default Matter event handler

markaj-nordic opened the pull request:
Implemented common Matter event handler module that includes the default handler and API to register user provided handlers in the application code.

This PR is for early stage review, so only lock sample has been reworked, until we agree upon suggested changes.

aasinclair committed:
samples: pmic: npm1300_one_button: Added sample

Added one button nPM1300 PMIC sample

Signed-off-by: Andy Sinclair <andy.sinclair@nordicsemi.no>

koffes committed:
Applications: nrf5340_audio: refactored sending iso data

Remove duplicate code for ISO TX.
OCT-2819.

Signed-off-by: Kristoffer Rist Skøien <kristoffer.skoien@nordicsemi.no>

kkasperczyk-no committed:
applications: Added BT security support for the Matter bridge

Extended the Matter bridge BLE connectivity manager by the
support for BT SMP and allowing to establish secure connections
with the Bluetooth LE bridged devices.

Signed-off-by: Kamil Kasperczyk <kamil.kasperczyk@nordicsemi.no>

rugeGerritsen committed:
Bluetooth: Controller: New default event length and spacing for ISO

When ISO is enabled, we want to set asside as much time as possible
for ISO activities.

By setting BT_CTLR_SDC_MAX_CONN_EVENT_LEN_DEFAULT=0, we ensure that
the ACL will utilize as little time as possible and avoid increasing
the data length capabilities.

The BT_CTLR_SDC_CENTRAL_ACL_EVENT_SPACING_DEFAULT needs to be updated
accordingly - we want the ACLs to be spaced apart to avoid scheduling
conflicts. By selecting the value of 30 ms, we ensure that:
 - The ACLs are non-overlapping given that they are using an interval
   which is greater than 30 ms.
 - The ACL events are not overlapping with ISO events, given that the
   ISO interval is a multiple of 7.5 ms or 10 ms.

Signed-off-by: Rubin Gerritsen <rubin.gerritsen@nordicsemi.no>

jayteemo committed:
net: lib: nrf_cloud: handle anchor type

Handle location request responses fulfilled by a Wi-Fi anchor.
IRIS-7940

Signed-off-by: Justin Morton <justin.morton@nordicsemi.no>

ankuns committed:
manifest: Update sdk-nrf-802154 and nrfxlib revisions

This commit updates sdk-nrf-802154 and nrfxlib revisions to bring in
the latest changes in the nrf_802154 component.

Signed-off-by: Andrzej Kuros <andrzej.kuros@nordicsemi.no>

kkasperczyk-no committed:
manifest: Pulled fix for the Matter over Thread DNS resolution

There was a bug discovered that in specific scenarios lead to
Matter over Thread device application crash while processing
DNS resolve response.

This PR pulls commit with a fix and adds known issues entry that
describes the problem.

Signed-off-by: Kamil Kasperczyk <kamil.kasperczyk@nordicsemi.no>

kkasperczyk-no commented on samples/matter/lock/CMakeLists.txt:
```diff
@@ -53,6 +53,7 @@ target_sources(app PRIVATE
     ${COMMON_ROOT}/src/tasks/task_executor.cpp
     ${COMMON_ROOT}/src/board/board.cpp
     ${COMMON_ROOT}/src/init/matter_init.cpp
+    ${COMMON_ROOT}/src/events/matter_event_handler.cpp
```
Would it make sense to move this file to already existing `tasks` or `init` directory? I mean, maybe it's not the best idea to create a new directory for each new pair of .h + .cpp files.

markaj-nordic replied:
Yep, as discussed f2f, I will fix that.

markaj-nordic replied:
Moved to matter_api directory together with Matter init utils.

kkasperczyk-no commented on samples/matter/common/src/events/matter_event_handler.h:
```diff
@@ -0,0 +1,51 @@
+	/**
+	 * @brief Safely register custom Matter event handler
+	 *
+	 * This function adds user provided Matter event handler in a thread safe manner.
```
Could you copy and paste part of `DefaultEventHandler` description about what Matter event is here? I think it would be good to briefly describe what kinds of events are generated and can be handled in the given handler.

markaj-nordic replied:
Sure.

markaj-nordic replied:
I added the generic documentation bit on top of the event handler module.

kkasperczyk-no commented on samples/matter/common/src/events/matter_event_handler.h:
```diff
@@ -0,0 +1,51 @@
+	/**
+	 * @brief Safely register custom Matter event handler
+	 *
+	 * This function adds user provided Matter event handler in a thread safe manner.
```
Adding custom handler will not automatically unregister `DefaultEventHandler`, right? Maybe it would be good to add one sentence that it extends the `DefaultEventHandler` functionality and it is not going to replace it (first I thought it is going to override it).

markaj-nordic replied:
Yes, we can register multiple handlers (these are kept as a linked list in the server). I will add a comment on that.

Damian-Nordic commented on samples/matter/common/src/init/matter_init.cpp:
```diff
@@ -209,7 +209,8 @@ namespace Matter
 		/* Schedule all CHIP initializations to the CHIP thread for better synchronization. */
-		return PlatformMgr().ScheduleWork(DoInitChipServer, reinterpret_cast<intptr_t>(matterEventHandler));
+		return PlatformMgr().ScheduleWork(DoInitChipServer,
+						  reinterpret_cast<intptr_t>(sLocalInitData.mEventHandler));
```
It's already in `sLocalInitData` so maybe not needed to be passed second time?

markaj-nordic replied:
Right. You mean using sLocalInitData directly in DoInitChipServer?

Damian-Nordic replied:
Yes

markaj-nordic replied:
Done.

Damian-Nordic commented on samples/matter/common/src/events/matter_event_handler.cpp:
```diff
@@ -0,0 +1,97 @@
+#endif /* CONFIG_CHIP_OTA_REQUESTOR */
+#endif /* CONFIG_NET_L2_OPENTHREAD */
+			if (isNetworkProvisioned) {
+				GetBoard().UpdateDeviceState(DeviceState::DeviceProvisioned);
```
I wonder if it wouldn't be clear if board registered its own event handler if needed. Right now the code is split into components which are interdependent so it may be a good idea to at some point end up with components that can be turned on independently.

markaj-nordic replied:
I was actually thinking about similar approach. I will try do address that.

kkasperczyk-no committed:
samples: matter: Added Nordic namespace to all common modules

The common modules delivered by Nordic should be clearly
separated from the generic code, so the namespace Nordic
was added to all of them.

By the way refactored directories structure a bit to group
files of similar purpose in a one location.

Signed-off-by: Kamil Kasperczyk <kamil.kasperczyk@nordicsemi.no>

Damian-Nordic reviewed (approved):
Approving assuming a followup on splitting the event handler. There's a clash with Kamil's PR regarding directory of `matter_init` module: https://github.com/nrfconnect/sdk-nrf/pull/13580 but you're probably aware of that :)

markaj-nordic commented:
> Approving assuming a followup on splitting the event handler. There's a clash with Kamil's PR regarding directory of `matter_init` module: [nrfconnect#13580](https://github.com/nrfconnect/sdk-nrf/pull/13580) but you're probably aware of that :)

@Damian-Nordic please take a look one more time.

markaj-nordic committed:
samples: matter: Unified default Matter event handler

Implemented common Matter event handler module that
includes the default handler and API to register user
provided handlers in the application code.

Added default Matter event handler for GPIO related operations
in the Board utility module, that can be registered
in the application.

Signed-off-by: Marcin Kajor <marcin.kajor@nordicsemi.no>

markaj-nordic commented:
Moved do sdk-nrf.

markaj-nordic commented:
Moved do sdk-nrf.
